build/Build.Steps.cs

```diff
@@ -532,19 +530,32 @@ partial class Build
                     depsJson.RemoveDuplicatedLibraries(architectureStores);
                     depsJson.RemoveOpenTelemetryLibraries();
 
+                    // To allow roll forward for applications, like Roslyn, that target one tfm
+                    // but have a later runtime move the libraries under the original tfm folder
+                    // to the latest one.
                     if (folderRuntimeName == TargetFramework.NET6_0)
                     {
-                        // To allow roll forward for applications, like Roslyn, that target one tfm
-                        // but have a later runtime move the libraries under the original tfm folder
-                        // to the latest one.
-                        depsJson.RollFrameworkForward(TargetFramework.NET6_0, TargetFramework.NET7_0, architectureStores);
+                        depsJson.RollFrameworkForward(TargetFramework.NET6_0, TargetFramework.NET8_0, architectureStores);
+                    }
+                    else if (folderRuntimeName == TargetFramework.NET7_0 || folderRuntimeName == TargetFramework.NET8_0)
+                    {
+                        depsJson.RollFrameworkForward(TargetFramework.NET7_0, TargetFramework.NET8_0, architectureStores);
                     }
 
                     // Write the updated deps.json file.
                     File.WriteAllText(file, depsJson.ToJsonString(new()
                     {
                         WriteIndented = true
                     }));
+
+                    // workaround for rc version
+                    if (file!.Parent!.Name.Contains("8.0.0"))
+                    {
+                        var newDirectory = file.Parent.Parent / "8.0.0-rc.1.23419.4";
```

Comment:
File stored in `/AdditionalDeps/shared/Microsoft.NETCore.App/8.0.0/OpenTelemetry.AutoInstrumentation.AdditionalDeps.deps.json,` is not handled correctly by dotnet runtime.

@rajkumar-rangaraj, could you please check with .NET team that it will be working with stable version of .NET and further patch releases? We should avoid crating multiple json files versions

Reply:
"It won't work with the GA version or even the next RC version. We can test using this approach, but before the release, we need to use the GA version of .NET 8.0 and rename the folder.